src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java

```diff
@@ -0,0 +1,213 @@
+ * definitions. The scope to verify is specified using the {@code Scope} class and defaults
+ * to {@code Scope.PRIVATE}. To verify another scope, set property scope to one of the
+ * {@code Scope} constants.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code scope} - specify the visibility scope where Javadoc comments are checked.
+ * Default value is {@code private}.
+ * </li>
+ * <li>
+ * Property {@code excludeScope} - specify the visibility scope where Javadoc comments are not
+ * checked. Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code allowedAnnotations} - specify the list of annotations that allow missed
+ * documentation. Only short names are allowed, e.g.{@code Generated}. Default value is
+ * {@code Generated}.
+ * </li>
+ * <li>
+ * Property {@code tokens} - tokens to check Default value is:
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INTERFACE_DEF">INTERFACE_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CLASS_DEF">CLASS_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF">ENUM_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF">ANNOTATION_DEF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the default check:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocType"/&gt;
+ * </pre>
+ * <p>
+ * To configure the check for {@code public} scope:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocType"&gt;
+ *   &lt;property name="scope" value="public"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * To configure the check for {@code private} classes only:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocType"&gt;
+ *   &lt;property name="scope" value="private"/&gt;
+ *   &lt;property name="excludeScope" value="package"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example that allows missing comments for classes annotated with
+ * {@code &#64;SpringBootApplication} and {@code &#64;Configuration}:
+ * </p>
+ * <pre>
+ * &#64;SpringBootApplication // no violations about missing comment on class
+ * public class Application {}
+ *
+ * &#64;Configuration // no violations about missing comment on class
+ * class DatabaseConfiguration {}
+ * </pre>
+ * <p>
+ * Use following configuration:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocType"&gt;
+ *   &lt;property name="allowedAnnotations" value="SpringBootApplication,Configuration"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * @since 8.19
+ */
+@StatelessCheck
+public class MissingJavadocTypeCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_JAVADOC_MISSING = "javadoc.missing";
+
+    /** Specify the visibility scope where Javadoc comments are checked. */
+    private Scope scope = Scope.PRIVATE;
```

Comment:
such default does not make sense.
Lets change it to "public", as in application development practice, only public methods are reasonable to check.

Reply:
Done.